### PR TITLE
Lift some utils code from the k/k repo to avoid relying on it

### PR DESCRIPTION
**Pull request by iawia002**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Directly relying on the k8s main repo will bring many problems. First, we need to add a lot of `replace` instructions to the `go.mod` file, and the k8s community does not encourage this. And we can no longer use `go install` to install `karmadactl` after we have `replace` instructions in our `go.mod` file.

So I looked at the code where we currently rely on the k8s repo and found that we don’t have a deep dependency on it, so I copied the utils code we depended on into the karmada repo and deleted the k8s dependency.

Ref https://github.com/karmada-io/karmada/pull/920#discussion_r744120051

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @Garrybest @RainbowMango

### CODE REVIEW

**Comment by Garrybest** [comment]
I'm afraid this PR may cause some problems when generating proto files because we have a dependency on api proto files of `k/k`. We can't remove the repo from `go.mod` directly. https://github.com/karmada-io/karmada/blob/5b068c31da1ae96e1d302b36404f44821f2415a7/hack/update-estimator-protobuf.sh#L29-L35

**Comment by iawia002** [comment]
> I'm afraid this PR may cause some problems when generating proto files because we have a dependency on api proto files of `k/k`. We can't remove the repo from `go.mod` directly.
> 
> https://github.com/karmada-io/karmada/blob/5b068c31da1ae96e1d302b36404f44821f2415a7/hack/update-estimator-protobuf.sh#L29-L35

Seems we didn't rely on the `k8s.io/kubernetes` repo, we just need `k8s.io/apimachinery` and `k8s.io/api`, I did some tests, it works well.

**Comment by Garrybest** [comment]
Oh, you're right. I got a mistake. Thanks for reminding, @iawia002. 😄

File: hack/verify-import-aliases.sh [hunk 1]
```diff
@@ -10,7 +10,39 @@ ROOT_PATH=$(pwd)
 
 temp_path=$(mktemp -d)
 pushd "${temp_path}" >/dev/null
-cp "${ROOT_PATH}"/go.mod .
+cat > go.mod <<EOF
```
Comment (iawia002) [inline hack/verify-import-aliases.sh:13]: NOTE: This is only a temporary solution and will be refactored in #920

**Review by iawia002 — COMMENTED** [review]
Hi @Garrybest @RainbowMango Do you have more comments?

**Comment by Garrybest** [comment]
Cool, I'm ok with the idea. Leave it to @RainbowMango for final decision.

**Comment by RainbowMango** [comment]
I'll look into it today.
/assign

**Comment by RainbowMango** [comment]
/lgtm
/approve

I also comment at https://github.com/karmada-io/karmada/pull/920#discussion_r755997896.

**Pull request merged**